core/launch_configuration.go

```diff
@@ -159,3 +150,43 @@ func copyBlockDeviceMappings(
 	}
 	return ec2BDMlist
 }
+
+// We don't know whether we got security group names or ids. We assume
+// that the ones starting with "sg-" are ids and then search for the IDs
+// of the other ones.
+func getSecurityGroupIDs(conn *connections, secGroups []*string) ([]*string, error) {
+	var (
+		names    []*string
+		ids      []*string
+		outNames *ec2.DescribeSecurityGroupsOutput
+		err      error
+	)
+
+	for _, secGroupStr := range secGroups {
+		// we assume these are IDs already
+		if strings.HasPrefix(*secGroupStr, "sg-") {
```

Comment:
Please make this check more strict, because the IDs have a certain format, fixed length and certain allowed characters so we could also use regex matching.

Reply:
Ok, I can use a regex to do that. Do you want me to add commits to this PR or just rebase my changes?

Reply:
Feel free to add more commits, I'll squash them all at merge time.